### PR TITLE
Feat product listing

### PR DESCRIPTION
**Pull request by aatishpandey**
Products Listing Page with Filters and Api call.

https://user-images.githubusercontent.com/32216307/161952189-34b22547-3ebf-4f6c-b6e1-2db9af86e38c.mp4

### CODE REVIEW

File: src/utils/filterCategory.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    if (category.keyboard) {
+        const filteredList = productList.filter(item => {
+            // checking rating
+            if (item.category === "keyboard") {
+                // checking if item already exists in final array
+                if (!finalList.some((finalItem) => item.id === finalItem.id)) {
+                    return item;
+                }
+            }
```
Comment (nishasen) [inline src/utils/filterCategory.js:27]: You can do these in one if else only, by using variable

File: src/utils/productReducerFn.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        case "KEYBOARD": return {
+            ...state,
+            category: { ...state.category, keyboard: !state.category.keyboard }
+        }
+
+        case "HEADPHONE": return {
+            ...state,
+            category: { ...state.category, headphone: !state.category.headphone }
```
Comment (nishasen) [inline src/utils/productReducerFn.js:49]: no need to have separate state here, you can change in one state only

**Review by nishasen — APPROVED** [review]
Code is clean and neatly written, in utils and reducer you can optimize a lot of code.

File: src/contexts/productContext.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            four: false,
+            three: false,
+            two: false,
+            one: false
+        },
+        category: {
+            monitor: false,
+            headphone: false,
+            keyboard: false,
+        },
```
Comment (krishanu2911) [inline src/contexts/productContext.js:20]: This could be optimized to only one no need to specify differently for different products

**Review by krishanu2911 — APPROVED** [review]
Everything is good to go👍🏻 No changes are required as of now 
But added a suggestion for optimising the reducer state.

**Pull request merged**